### PR TITLE
pyrex: enable containers-by-digest

### PR DESCRIPTION
**Pull request by amstewart**
The current pyrex implementation assumes that users wish to start
containers using the `$name:$tag` format. Users may not use the
`$name@$digest` format, which is preferable in environments where
build-reproducibility is required.

Change the pyrex.ini to eschew this assumption and update the init
script to allow the user to pull containers by digest.

Signed-off-by: Alex Stewart <alex.stewart@ni.com>

----

This PR is needed to use static containers in our AZDO pipeline.

# Testing
* Exported a `PYREX_BUILD_NILRT_IMAGE` variable with the full digest name of the latest `hardknott`-tagged `build-nilrt` container on `rnd-builds.repos.natinst.com`, and ran the init script. It correctly used the container-by-digest.
* Unexported the image variable and ran the initscript. Pyrex started using the `build-nilrt:latest` tag from my dev-machine.

### CODE REVIEW

**Comment by amstewart** [comment]
# Patch V2
* commit spelling.

**Review by shruthi-ravi — APPROVED** [review]
Do you want to hold this change until the build is stable? I suppose there is an associated azdo change too that needs to go in.

File: ni-oe-init-build-env
Comment (shruthi-ravi) [inline ni-oe-init-build-env:136]: Once a release branch is created, should the default be updated to the respective release-tagged container, like what you are doing in the sumo changelist.

Comment (shruthi-ravi) [reply ni-oe-init-build-env:136]: Or why not use the same logic used for the sumo branch?

Comment (amstewart) [reply ni-oe-init-build-env:136]: I have bigger plans for setting and using the codename in hardknott, as well as more changes to simplify the pyrex configuration. But I wanted to get these MVP changes merged first, so that I can unblock the work to switch to using static container IDs in AZDO.

**Comment by amstewart** [comment]
> Do you want to hold this change until the build is stable? I suppose there is an associated azdo change too that needs to go in.

I'm OK putting it in now. I believe the only remaining current issue with the pipelines affects only the `nilrt_os_common` component, which won't be affected by these changes. And getting these in will unblock my AZDO work and let me start PR validation runs.

**Comment by shruthi-ravi** [comment]
> I'm OK putting it in now. I believe the only remaining current issue with the pipelines affects only the nilrt_os_common component, which won't be affected by these changes. And getting these in will unblock my AZDO work and let me start PR validation runs.

I'm less confident with pulling additional changes while the build is already down :). But I'll leave the decision to you.

**Pull request merged**
